From 653a0742a8cf280f49832efcf4bad065330f5cee Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Wed, 7 Nov 2018 17:21:41 +0100
Subject: [PATCH 04/11] libstagefright: omx: Add support for loading prebuilt
 ddp decoder lib

* This lib additionally supports loading EAC3{,-JOC}, however
  I have never seen anything other than AC3 in the wild

Change-Id: I376211ddb8a518a93fdc64bc4b71d476d1f26479
---
 media/libstagefright/omx/SoftOMXPlugin.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libstagefright/omx/SoftOMXPlugin.cpp b/media/libstagefright/omx/SoftOMXPlugin.cpp
index 1f3e8c17b..27af3eddd 100644
--- a/media/libstagefright/omx/SoftOMXPlugin.cpp
+++ b/media/libstagefright/omx/SoftOMXPlugin.cpp
@@ -28,6 +28,9 @@
 
 namespace android {
 
+// Symbol needed to make libstagefright_soft_ddpdec load
+const char *MEDIA_MIMETYPE_AUDIO_EAC3_JOC = "audio/eac3-joc";
+
 static const struct {
     const char *mName;
     const char *mLibNameSuffix;
@@ -66,6 +69,7 @@ static const struct {
     { "OMX.google.flac.decoder", "flacdec", "audio_decoder.flac" },
     { "OMX.google.flac.encoder", "flacenc", "audio_encoder.flac" },
     { "OMX.google.gsm.decoder", "gsmdec", "audio_decoder.gsm" },
+    { "OMX.dolby.ac3.decoder", "ddpdec", "audio_decoder.ac3" },
 };
 
 static const size_t kNumComponents =
-- 
2.17.1

